integration/containerd/cri/integration-tests.sh

```diff
@@ -319,6 +342,135 @@ TestContainerMemoryUpdate() {
 	testContainerStop
 }
 
+TestContainerSwap() {
+	if [[ "${KATA_HYPERVISOR}" != "qemu" ]] || [[ "${ARCH}" != "x86_64" ]]; then
+		return
+	fi
+
+	local container_yaml=${REPORT_DIR}/container.yaml
+
+	info "Test container with guest swap"
+
+	create_containerd_config "${containerd_runtime_test}" 1
+	sudo sed -i -e 's/^#enable_guest_swap.*$/enable_guest_swap = true/g' "${kata_config}"
+
+	# Test without swap device
+	testContainerStart 0
+	swappiness=$(crictl exec $cid cat /sys/fs/cgroup/memory/memory.swappiness)
+	# Current default swappiness is 60
+	if [ $swappiness -ne 60 ]; then
+		testContainerStop
+		die "The VM swappiness $swappiness without swap device is not right"
+	fi
+	swap_in_bytes=$(crictl exec $cid cat /sys/fs/cgroup/memory/memory.memsw.limit_in_bytes)
+	if [ $swap_in_bytes -lt 1125899906842624 ]; then
+		testContainerStop
+		die "The VM swap_in_bytes $swap_in_bytes without swap device is not right"
+	fi
+	swap_size=$(($(crictl exec $cid cat /proc/meminfo | grep "SwapTotal:" | awk '{print $2}')*1024))
+	if [ $swap_size -ne 0 ]; then
+		testContainerStop
+		die "The VM swap size $swap_size without swap device is not right"
+	fi
+	testContainerStop
+
+	# Test with swap device
+	cat << EOF > "${container_yaml}"
+metadata:
+  name: busybox-killed-vmm
+annotations:
+  io.katacontainers.container.resource.swappiness: "100"
+  io.katacontainers.container.resource.swap_in_bytes: "1610612736"
+linux:
+  resources:
+    memory_limit_in_bytes: 1073741824
+image:
+  image: "$image"
+command:
+- top
+EOF
+	testContainerStart 1
+	swappiness=$(crictl exec $cid cat /sys/fs/cgroup/memory/memory.swappiness)
+	if [ $swappiness -ne 100 ]; then
+		testContainerStop
+		die "The VM swappiness $swappiness with swap device is not right"
+	fi
+	swap_in_bytes=$(crictl exec $cid cat /sys/fs/cgroup/memory/memory.memsw.limit_in_bytes)
+	if [ $swap_in_bytes -ne 1610612736 ]; then
+		testContainerStop
+		die "The VM swap_in_bytes $swap_in_bytes with swap device is not right"
+	fi
+	swap_size=$(($(crictl exec $cid cat /proc/meminfo | grep "SwapTotal:" | awk '{print $2}')*1024))
+	if [ $swap_size -ne 536870912 ]; then
+		testContainerStop
+		die "The VM swap size $swap_size with swap device is not right"
+	fi
+	testContainerStop
+
+	# Test without swap_in_bytes
+	cat << EOF > "${container_yaml}"
+metadata:
+  name: busybox-killed-vmm
+annotations:
+  io.katacontainers.container.resource.swappiness: "100"
+linux:
+  resources:
+    memory_limit_in_bytes: 1073741824
+image:
+  image: "$image"
+command:
+- top
+EOF
+	testContainerStart 1
+	swappiness=$(crictl exec $cid cat /sys/fs/cgroup/memory/memory.swappiness)
+	if [ $swappiness -ne 100 ]; then
+		testContainerStop
+		die "The VM swappiness $swappiness without swap_in_bytes is not right"
+	fi
+	swap_in_bytes=$(crictl exec $cid cat /sys/fs/cgroup/memory/memory.memsw.limit_in_bytes)
+	if [ $swap_in_bytes -lt 1125899906842624 ]; then
```

Comment:
I didn't get that number. Shouldn't swap_in_bytes be `2 * memory_limit_in_bytes`?

Reply:
swap_in_bytes is not set, it should be a value that bigger than 1125899906842624.